### PR TITLE
Feature/saved palette dragndrop

### PR DESCRIPTION
**Pull request by aspitz1**
I refactored the saved palette drag and drop to use parent element to target the id on the wrapper div so there isn't a need to have the data-id on all the divs. Added grab and grabbing cursor. Still doesn't persist on drag (known issue).

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -111,26 +111,55 @@ function dragoverHandler(event) {
 
 function dropHandler(event) {
   event.preventDefault();
-  var dragId = parseInt(event.dataTransfer.getData('color-box'));
+  var dragId = parseInt(event.dataTransfer.getData("color-box"));
+  var dropId = parseInt(event.target.dataset.id);
   var dropId = parseInt(event.target.parentNode.dataset.id);
```
Comment (tfettig22) [inline main.js:119]: I noticed we have 2 declarations of var dropId here, do we need both?

File: main.js [hunk 2]
```diff
@@ -111,26 +111,55 @@ function dragoverHandler(event) {
...
-        dragIndex = i;
-      }
-      if(currentPalette.colors[i].id === dropId){
-        dropColor = currentPalette.colors[i];
-        dropIndex = i;
-      }
+  for(var i = 0; i < currentPalette.colors.length; i++) {
+    if(currentPalette.colors[i].id === dragId){
+      dragColor = currentPalette.colors[i];
+      dragIndex = i;
```
Comment (tfettig22) [inline main.js:127]: These changes seem to be the same code, unless I'm missing something? Maybe just a syntax issue?

Comment (aspitz1) [reply main.js:127]: It was the syntax issue.

File: main.js (hunk 1 above)
Comment (aspitz1) [reply main.js:119]: Idk how that got there. I don't think we need line 115. I'll fix it and push it.

I took off and tested it and as far as I can see everything works fine.

**Review by tfettig22 — APPROVED** [review]

**Comment by dstavis** [comment]
I'd prefer to wait until we fix the saved-palette-to-color-container dragndrop bug before merging this. Even if it's unlikely, I wouldn't want to have a version of main that a user could trigger a bug on.

**Review by tfettig22 — APPROVED** [review]

**Review by dstavis — APPROVED** [review]

**Pull request merged**